test/AxelarGateway.js

```diff
@@ -101,14 +102,58 @@ describe('AxelarGateway', () => {
 
         const params = getWeightedProxyDeployParams(governance.address, mintLimiter.address, [], [], threshold);
 
-        const proxy = await gatewayProxyFactory.deploy(gatewayImplementation.address, params);
-        await proxy.deployTransaction.wait(network.config.confirmations);
+        gatewayProxy = await gatewayProxyFactory.deploy(gatewayImplementation.address, params);
+        await gatewayProxy.deployTransaction.wait(network.config.confirmations);
 
-        await auth.transferOwnership(proxy.address).then((tx) => tx.wait(network.config.confirmations));
+        await auth.transferOwnership(gatewayProxy.address).then((tx) => tx.wait(network.config.confirmations));
 
-        gateway = gatewayFactory.attach(proxy.address);
+        gateway = gatewayFactory.attach(gatewayProxy.address);
     };
 
+    describe('axelar gateway proxy', () => {
+        it('should revert on invalid gateway implementation address', async () => {
+            const params = '0x';
+
+            await expectRevert(
+                (gasOptions) => gatewayProxyFactory.deploy(AddressZero, params, gasOptions),
+                gatewayProxyFactory,
+                'InvalidImplementation',
+            );
+        });
+
+        it('should revert if gateway setup fails', async () => {
+            const params = '0x00';
+            const operatorAddresses = getAddresses(operators);
+
+            auth = await authFactory.deploy(getWeightedAuthDeployParam([operatorAddresses], [getWeights(operatorAddresses)], [threshold]));
+            await auth.deployTransaction.wait(network.config.confirmations);
+
+            const gatewayImplementation = await gatewayFactory.deploy(auth.address, auth.address);
+            await gatewayImplementation.deployTransaction.wait(network.config.confirmations);
+
+            await expectRevert(
```

Comment:
can you use `expectRevert` for custom revert tests as well? it's needed for some live networks

Reply:
Changed it across the entire repo